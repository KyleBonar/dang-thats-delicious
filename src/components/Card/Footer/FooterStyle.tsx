import styled from "../../../theme/styled-components";
import { Link } from "../../Link/Link";

export const StyledLink = styled(Link)`
  button {
    text-transform: uppercase;
    font-size: 14px;
    letter-spacing: 1px;
    margin: auto auto 0;
  }
`;
StyledLink.displayName = "Link";
