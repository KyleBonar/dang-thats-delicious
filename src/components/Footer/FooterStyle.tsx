import styled from "../../theme/styled-components";

export const StyledFooter = styled.footer`
  padding: 1em;
  width: 100%;
  background-color: ${props => props.theme.primaryThemeColor};
`;

export const StyledDiv = styled.div`
  max-width: ${props => props.theme.maxPageWidth};
  margin: 0 auto;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  flex-wrap: wrap;

  > div {
    width: 50%;
    box-sizing: border-box;
    padding: 0 1em;
  }

  > div:last-of-type {
    width: 100%;
  }

  @media (min-width: ${props => props.theme.smToMd}) {
    flex-wrap: nowrap;

    > div,
    > div:last-of-type {
      width: 33%;
    }
  }
`;
