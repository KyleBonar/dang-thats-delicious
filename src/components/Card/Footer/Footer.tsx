import * as React from "react";

import { Button } from "../../Button/Button";
import { StyledLink } from "./FooterStyle";

export interface IFooterProps {
  to: string;
  anchorText?: string;
}

const Footer: React.FunctionComponent<IFooterProps> = props => {
  const { to, anchorText = "View" } = props;

  return (
    <StyledLink href={to}>
      <Button displayType="outline">{anchorText}</Button>
    </StyledLink>
  );
};

export default Footer;
