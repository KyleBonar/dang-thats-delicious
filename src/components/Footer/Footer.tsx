import * as React from "react";

import Navigation from "./components/Navigation/Navigation";
import Types from "./components/Types/Types";
import About from "./components/About/About";
import { StyledFooter, StyledDiv } from "./FooterStyle";

interface FooterProps {}

const Footer: React.FC<FooterProps> = props => {
  return (
    <StyledFooter>
      <StyledDiv>
        <Navigation />
        <Types />
        <About />
      </StyledDiv>
    </StyledFooter>
  );
};

export default Footer;
