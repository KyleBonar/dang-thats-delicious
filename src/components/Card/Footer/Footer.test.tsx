import * as React from "react";
import * as enzyme from "enzyme";

import Footer from "./Footer";

describe("<Footer />", () => {
  it("renders", () => {
    const wrapper = enzyme.shallow(<Footer to="" />);

    expect(wrapper).toBeTruthy();
  });
});
