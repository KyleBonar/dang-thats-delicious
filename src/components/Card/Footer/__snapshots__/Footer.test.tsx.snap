// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> matches snapshot 1`] = `
<Link
  href=""
>
  <Styled(Button)
    displayType="outline"
  >
    View
  </Styled(Button)>
</Link>
`;
